android/app/src/main/java/safetyfirst/androidapp/safetyfirstcontroller/fragments/LoginFragment.java

```diff
@@ -0,0 +1,123 @@
+
+import java.util.Objects;
+
+import safetyfirst.androidapp.safetyfirstcontroller.MainActivity;
+import safetyfirst.androidapp.safetyfirstcontroller.R;
+
+public class LoginFragment extends Fragment {
+    private EditText editTextMail, editTextPass;
+    private Button logIn;
+    private TextView register;
+
+    private FirebaseAuth firebaseAuth;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_login, container, false);
+
+        setHasOptionsMenu(true);
+
+        firebaseAuth = FirebaseAuth.getInstance();
+
+        TextView register = rootView.findViewById(R.id.toRegister);
+        Button logIn = rootView.findViewById(R.id.loginUser);
+
+        EditText editTextMail = rootView.findViewById(R.id.loginEmail);
+        EditText editTextPass = rootView.findViewById(R.id.loginPassword);
+
+        register.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                FragmentTransaction addTransaction = getParentFragmentManager().beginTransaction();
+                addTransaction.replace(R.id.fragment_container, new RegisterFragment());
+                addTransaction.commit();
+            }
+        });
+
+        logIn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String email = editTextMail.getText().toString().trim();
+                String password = editTextPass.getText().toString().trim();
+
+                if (email.isEmpty()) {
+                    editTextMail.setError("Email is required");
+                    editTextMail.requestFocus();
+                    return;
+                }
+                if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
+                    editTextMail.setError("Enter a valid Email");
+                    editTextMail.requestFocus();
+                    return;
+                }
+                if (password.isEmpty()) {
+                    editTextPass.setError("Password is required");
+                    editTextPass.requestFocus();
+                    return;
+                }
+                firebaseAuth.signInWithEmailAndPassword(email, password)
+                        .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                            @Override
+                            public void onComplete(@NonNull Task<AuthResult> task) {
+                                if (task.isSuccessful()) {
+                                    hideKeyboardFrom(requireContext(), view);
+
+                                    NavigationView navigationView = (NavigationView) requireActivity().findViewById(R.id.nav_view);
+                                    Menu nav_Menu = navigationView.getMenu();
```

Comment:
naming conventions

Reply:
Refer to previously resolved conversation.

Reply:
Same again

Reply:
Fixed. Thanks.